src/lib/y2security/lsm/config.rb

```diff
@@ -0,0 +1,167 @@
+
+      def initialize
+        supported.each { |m| self.class.send(:define_method, m.id) { m } }
+      end
+
+      # Select the LSM to be used based in the one defined in the control file using apparmor as
+      # fallback in case that no one is selected
+      def propose_default
+        log.info("The settings are #{product_feature_settings.inspect}")
+        selected = product_feature_settings.fetch(:select, "apparmor")
+
+        select(selected)
+      end
+
+      # Select the module with the given id when it is selectable.
+      #
+      # @param id [String,Symbol] LSM id to be selected
+      def select(id)
+        @selected = selectable.find { |m| m.id == id.to_sym }
+        log.info @selected ? "Selected LSM with id: #{id}" : "No LSM selected with id: #{id}"
+        @selected
+      end
+
+      # Convenience method to obtain a list of the supported and selectable modules
+      #
+      # @return [Array<Y2Security::LSM::Base] array of supported and selectable LSMs
+      def selectable
+        supported.select(&:selectable?)
+      end
+
+      # Returns the needed patterns for the selected LSM or an empty array if no one is selected
+      #
+      # @return [Array<Sting>]
+      def needed_patterns
+        return [] unless selected
+
+        selected.needed_patterns
+      end
+
+      # Convenience method to save the configuration for the selected LSM
+      def save
+        return false unless selected
+
+        supported.each(&:reset_kernel_params)
+        selected.save
+      end
+
+      # Obtains the supported and active Linux Security Major Module from the running system
+      #
+      # @return [Y2Security::LSM::Base]
+      def from_system
+        active.first
+      end
+
+      # Reads which Linux Security module is active and its configuration if needed but only in a
+      # running system as during the installation there is no active module at all returning false
+      # in that case
+      #
+      # @return [Boolean] whether the configuration was read or not
+      def read
+        return false unless Yast::Stage.normal
+
+        @selected = from_system
+        return false unless @selected
+
+        @selected.read
+      end
+
+      # Return an array with the supported and active Linux Security Major Modules
+      #
+      # @return [Array<Y2Security::LSM::Base>]
+      def active
+        return [] unless Yast::Stage.normal
+
+        modules = Yast::SCR.Read(Yast.path(".target.string"), RUNNING_PATH)
+        modules.split(",").each_with_object([]) do |name, result|
+          supported_module = supported.find { |m| m.id.to_s == name }
+          result << supported_module if supported_module
+        end
+      end
+
+      # Returns whether the LSM is configurable during installation or not based in the control file
+      # declaration. It returns false in case it is WSL
+      #
+      # @return [Boolean] true if LSM is configurable during the installation; false otherwise
```

Comment:
```suggestion
      # @return [Boolean] false when running in a WSL
      #                   whether LSM is configurable during the installation or not otherwise.
```